test/Rings-conformance-tests.jl

```diff
@@ -1,1041 +1,28 @@
-          b = swap_cols(a, 1, 2)
-          @test b == matrix(R, [2 1 ; 4 3])
-          @test a == matrix(R, [1 2 ; 3 4])
-
-          # TODO: reverse_rows, reverse_cols
-          # TODO: add_column, add_row
-          # TODO: multiply_column, multiply_row
-          # TODO: ! variants (such as `swap_cols!` etc.) of all of the above
-      end
-
-   end
-
-   return nothing
-end
-
-function test_MatAlgebra_interface(S::MatRing; reps = 20)
-
-   ST = elem_type(S)
-   R = base_ring(S)
-   T = elem_type(R)
-
-   @test nrows(S) == ncols(S)
-
-   @testset "MatRing interface for $(S) of type $(typeof(S))" begin
-
-      test_NCRing_interface(S, reps = reps)
-
-      @testset "Constructors" begin
-         for k in 1:reps
-            a = test_elem(S)::ST
-            @test nrows(a) == nrows(S)
-            @test ncols(a) == ncols(S)
-            @test a == S(T[a[i, j] for i in 1:nrows(a), j in 1:ncols(a)])
-            @test a == S(T[a[i, j] for i in 1:nrows(a) for j in 1:ncols(a)])
-         end
-      end
-
-      @testset "Basic manipulation of matrices" begin
-         for k in 1:reps
-            a = test_elem(S)::ST
-            A = deepcopy(a)
-            b = zero(S)
-            for i in 1:nrows(a), j in 1:ncols(a)
-               b[i, j] = a[i, j]
-            end
-            @test b == a
-            @test transpose(transpose(a)) == a
-            @test a == A
-         end
-      end
-
-      @testset "Determinant" begin
-         for k in 1:reps
-            a = test_elem(S)::ST
-            b = test_elem(S)::ST
-            A = deepcopy(a)
-            B = deepcopy(b)
-            @test det(a*b) == det(a)*det(b)
-            @test a == A
-            @test b == B
-         end
-      end
-   end
-
-   return nothing
-end
-
-function test_Ring_interface_recursive(R::AbstractAlgebra.Ring; reps = 50)
-   test_Ring_interface(R; reps = reps)
-   Rx, _ = polynomial_ring(R, :x)
-   test_Poly_interface(Rx, reps = 2 + fld(reps, 2))
-   Rxy, _ = polynomial_ring(R, [:x, :y])
-   test_MPoly_interface(Rxy, reps = 2 + fld(reps, 2))
-   S = matrix_ring(R, rand(0:3))
-   test_MatAlgebra_interface(S, reps = 2 + fld(reps, 2))
-   S = matrix_space(R, rand(0:3), rand(0:3))
-   test_MatSpace_interface(S, reps = 2 + fld(reps, 2))
-end
-
-function test_Field_interface_recursive(R::AbstractAlgebra.Field; reps = 50)
-   test_Ring_interface_recursive(R, reps = reps)
-   test_Field_interface(R, reps = reps)
-end
+# This file is just for backwards compatibility and should be removed in the future.
+
+# Assumptions (as that was already required with the previous setup):
+# - `AbstractAlgebra` is already loaded
+# - `Test` is available in the environment
+
+# load TestExt
+using Test
+
+import .ConformanceTests: generate_element as test_elem
+import .ConformanceTests: adhoc_partner_rings as adhoc_partner_rings
+import .ConformanceTests: equality as equality
+
+test_mutating_op_like_zero = ConformanceTests.test_mutating_op_like_zero
```

Comment:
Why not
```suggestion
import .ConformanceTests: test_mutating_op_like_zero
```
etc.?

Reply:
just lazyness. I used to have no imports at all here, but all functions that get extended downstream need an import. will change the rest as well